studio/components/interfaces/Organization/OAuthApps/RevokeAppModal.tsx

```diff
@@ -0,0 +1,73 @@
+import { useParams } from 'common'
+import { useAuthorizedAppRevokeMutation } from 'data/oauth/authorized-app-revoke-mutation'
+import { AuthorizedApp } from 'data/oauth/authorized-apps-query'
+import { useStore } from 'hooks'
+import { Alert, IconAlertOctagon, IconLock, Modal } from 'ui'
+
+export interface RevokeAppModalProps {
+  selectedApp?: AuthorizedApp
+  onClose: () => void
+}
+
+const RevokeAppModal = ({ selectedApp, onClose }: RevokeAppModalProps) => {
+  const { ui } = useStore()
+  const { slug } = useParams()
+  const { mutateAsync: revokeAuthorizedApp, isLoading: isDeleting } =
+    useAuthorizedAppRevokeMutation()
+
+  const onConfirmDelete = async () => {
+    if (!slug) return console.error('Slug is required')
+    if (!selectedApp?.id) return console.error('App ID is required')
+
+    try {
+      await revokeAuthorizedApp({ slug, id: selectedApp?.id })
+      ui.setNotification({
+        category: 'success',
+        message: `Successfully revoked the app "${selectedApp?.name}"`,
+      })
+      onClose()
+    } catch (error) {
+      ui.setNotification({
+        category: 'error',
+        message: `Failed to revoke app: ${(error as any).message}`,
+      })
+    }
+  }
+
+  return (
+    <Modal
+      size="medium"
+      alignFooter="right"
+      header={`Confirm to revoke ${selectedApp?.name}`}
+      visible={selectedApp !== undefined}
+      loading={false} // [TO UPDATE]
```

Comment:
to update now or later?

Reply:
oh you saved the dashboard from a UI bug right here

Reply:
something something code review